topologytest/couchbase_server_peer_test.go

```diff
@@ -174,11 +206,83 @@ func (p *CouchbaseServerPeer) CreateReplication(passivePeer Peer, config PeerRep
 }
 
 // SourceID returns the source ID for the peer used in <val>@sourceID.
-func (r *CouchbaseServerPeer) SourceID() string {
-	return r.sourceID
+func (p *CouchbaseServerPeer) SourceID() string {
+	return p.sourceID
 }
 
 // GetBackingBucket returns the backing bucket for the peer.
 func (p *CouchbaseServerPeer) GetBackingBucket() base.Bucket {
 	return p.bucket
 }
+
+// TB returns the testing.TB for the peer.
+func (p *CouchbaseServerPeer) TB() testing.TB {
+	return p.tb
+}
+
+// getDocVersion returns a DocVersion from a cas and xattrs with _vv (hlv) and _sync (RevTreeID).
+func getDocVersion(peer Peer, cas uint64, xattrs map[string][]byte) rest.DocVersion {
+	docVersion := rest.DocVersion{}
+	hlvBytes, ok := xattrs[base.VvXattrName]
+	if ok {
+		var hlv *db.HybridLogicalVector
+		require.NoError(peer.TB(), json.Unmarshal(hlvBytes, &hlv))
+		docVersion.CV = db.Version{SourceID: hlv.SourceID, Value: hlv.Version}
+	} else {
+		docVersion.CV = db.Version{SourceID: peer.SourceID(), Value: cas}
+	}
+	sync, ok := xattrs[base.SyncXattrName]
```

Comment:
RevTreeID will not (necessarily) be consistent across XDCR replications, so from that perspective I might have expected a CouchbaseServerPeer to ignore SyncXattr/RevTreeID as those are concepts only relevant when talking about SGW and CBL peers.  Are there some test cases that require this information to be populated?

Reply:
I think I was populating it so it would be present, but not necessarily using it for anything at this time. You could imagine that CBL would set this value, and then Couchbase Server would read it. It is helpful for debugging to know what revid it was, because sometimes I can correlate with where that came from, in addition to Source ID.

I think that there could/will be a need to add `PV` to this version of certain types of assertions, not yet implemented.